x/kavamint/abci_test.go

```diff
@@ -0,0 +1,196 @@
+package kavamint_test
+
+import (
+	"testing"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/cosmos/cosmos-sdk/x/staking"
+	"github.com/stretchr/testify/suite"
+
+	communitytypes "github.com/kava-labs/kava/x/community/types"
+	"github.com/kava-labs/kava/x/kavamint"
+	"github.com/kava-labs/kava/x/kavamint/keeper"
+	"github.com/kava-labs/kava/x/kavamint/testutil"
+	"github.com/kava-labs/kava/x/kavamint/types"
+)
+
+type abciTestSuite struct {
+	testutil.KavamintTestSuite
+}
+
+func (suite *abciTestSuite) SetupTest() {
+	suite.KavamintTestSuite.SetupTest()
+}
+
+func (suite abciTestSuite) CheckModuleBalance(ctx sdk.Context, moduleName string, expectedAmount sdk.Int) {
+	denom := suite.StakingKeeper.BondDenom(ctx)
+	amount := suite.App.GetModuleAccountBalance(ctx, moduleName, denom)
+	suite.Require().Equal(expectedAmount, amount)
+}
+
+func (suite *abciTestSuite) CheckFeeCollectorBalance(ctx sdk.Context, expectedAmount sdk.Int) {
+	suite.CheckModuleBalance(ctx, authtypes.FeeCollectorName, expectedAmount)
+}
+
+func (suite *abciTestSuite) CheckKavamintBalance(ctx sdk.Context, expectedAmount sdk.Int) {
+	suite.CheckModuleBalance(ctx, types.ModuleName, expectedAmount)
+}
+
+func (suite *abciTestSuite) CheckCommunityPoolBalance(ctx sdk.Context, expectedAmount sdk.Int) {
+	suite.CheckModuleBalance(ctx, communitytypes.ModuleAccountName, expectedAmount)
+}
+
+func TestABCITestSuite(t *testing.T) {
+	suite.Run(t, new(abciTestSuite))
+}
+
+func (suite *abciTestSuite) Test_BeginBlocker_MintsExpectedTokens() {
```

Comment:
agreed. i didnt have unhappy tests because the panics shouldnt be reachable. but i'll be sure they get tested when i refector the begin blocker to be one method (`AccumulateInterest`)